src/store/reducers/appReducer.ts

```diff
@@ -0,0 +1,38 @@
+export enum requestStatus {
+  failed = 'failed',
+  idle = 'idle',
+  loading = 'loading',
+  succeeded = 'succeeded',
+}
+
+const initialState = {
+  status: requestStatus.idle,
+};
+
+type initialStateType = {
+  status: RequestStatusType;
+};
+
+export type RequestStatusType =
+  | requestStatus.idle
+  | requestStatus.loading
+  | requestStatus.succeeded
+  | requestStatus.failed;
+
+export const appReducer = (
+  state: initialStateType = initialState,
+  action: ActionsType,
+) => {
+  switch (action.type) {
+    case 'APP/SET_STATUS':
+      return { ...state, status: action.status };
+
+    default:
+      return state;
+  }
+};
+
+export const toggleAppStatusAC = (status: RequestStatusType) =>
+  ({ type: 'APP/SET_STATUS', status } as const);
```

Comment:
create a separate folder called `actions` with actions inside
avoid `as const`, move status inside payload field and createa a separate type for this action which will return `{ type: typeof CONSTS.APP_SET_STATUS, payload: { status: typeof requestStatus } }`